Early-init support, aka chemacs 2.0

plexus opened the pull request:
Hi @emacs18 , I've merged your [last PR](https://github.com/plexus/chemacs/pull/40) into this repo and taken it from there. Instead of calling `chemacs-check-command-line-args` from init.el / early-init.el separately, I've put it back into `chemacs.el` (renamed from `chemacs-common.el`), and instead I'm communicating early init or not via a variable.

A little less elegant perhaps, but it means we can continue to support the old way of installing chemacs in `~/.emacs`.

I haven't given this a good spin yet but I think it's a good foundation. Still a bunch of things to be wrapped here

- [x] update README.md
- [x] remove install scripts
- [x] remove install instructions from the script, just refer to the README.
- [x] replace the auto-creation of `~/.emacs-profiles.el` with a warning telling people how to create it themselves. Less magic, more explicit, since there is no longer an obvious default for us to use.

Thinking this through maybe we should actually just remove the install scripts, and recommend people to clone this repo directly into `~/.emacs.d`. We can mention in the docs that people can alternatively symlink to `~/.emacs` if they don't need early-init.el, but in that case they might as well stick to chemacs 1.

Does that make sense? Thanks a lot for pusing this forward.

plexus commented:
Help with all of the above would greatly be appreciated. I think we can simplify things a lot, but we need clear instructions for people to follow, and need to test this to make sure it all works reliably.

emacs18 commented:
I started using this as my `~/.emacs.d`.  With it I have the following profiles working:
- spacemacs
- doom
- my setup
- spacemacs + my setup
- doom + my setup
I decided to create a package simply to house my own private setup.
That seemed to be easiest way to use that on its own or along with other setups.
So far so good.  I'll certainly update this project with any feedback I might have in the future.

plexus commented:
I've updated the installation instructions and changed the script so it no longer auto-creates .emacs-profile.el. Probably still some improvements to polish this up but I think that addresses the main issues.

My main concern here is that there's more to mess up. E.g. people still having a `~/.emacs` file, or putting chemacs in `~/.emacs.d`, but then pointing `~/.emacs-profiles.el` at `~/.emacs.d` creating an infinite loop. Might be good to add an install script again in the future that actually clones the repo to `~/.emacs.d` while checking for all these things.
